hw3/Ivaniv/task1.js

```diff
@@ -0,0 +1,106 @@
+const SarahMay = {
+	firstName: "Sarah",
+	lastName: "May",
+	birthDate: "13.03.1990",
+	address: {
+		country: "Spain",
+		city: "Seville",
+		street: "calle Verde",
+		house: "5",
+		apartment: "2A",
+	},
+	job: {
+		title: "teacher",
+		experience: 5
+	}
+}
+
+const JeremieBrown = {
+	firstName: "Jeremie",
+	lastName: "Brown",
+	birthDate: "10.11.1983",
+	address: {
+		country: "Great Britain",
+		city: "Bristol",
+		street: "Park street",
+		house: "14"
+	},
+	job: {
+		title: "sale manager",
+		experience: 10
+	}
+}
+
+const Elizabeth = {
+	firstName: "Elizabeth",
+	birthDate: "27.02.1998",
+	address: {
+		country: "Italy",
+		city: "Verona",
+		street: "via Calchirelli",
+		house: "9F",
+		apartment: "49",
+	},
+	job: {
+		title: "fitness coach",
+		experience: 1.5
+	}
+}
+
+class PersonClass {
+	constructor(person) {
+		this.person = person;
+	}
+
+	getFullName = () => {
+		var fullName;
+		for (const key in this.person) {
```

Comment:
weird approach... you could just check whether the needed properties present in obj without this loop.
```
const {firstName, lastName} = this.person;
if(!firstName) {
  return "No firstname!" // or throw an error
} else if (firstName && lastName) {
  return `${firstName} ${lastName}`
} else if (firstName && !lastName) {
  return firstName
}
```